File: 349-intersection-of-two-arrays/intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        set1 = set(nums1)
        set2 = set(nums2)
        
        # Find the smaller set to iterate through
        smaller_set = set1 if len(set1) < len(set2) else set2
        larger_set = set2 if smaller_set == set1 else set1
        
        result_set = set()
        
        # Iterate through the smaller set
        for num in smaller_set:
            # Check if the element exists in the other set
            if num in larger_set:
                result_set.add(num)
        
        return list(result_set)

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        sort(nums1.begin(),nums1.end());
        sort(nums2.begin(),nums2.end());
        vector<int>res;
        int left = 0,right =0;
        while(left<nums1.size() && right<nums2.size()){
            if(nums1[left] == nums2[right]){
            if(res.empty() || res.back() != nums1[left]){
                res.push_back(nums1[left]);
            }
            left++;
            right++;
            }
            else if (nums1[left] < nums2[right]){
                left++;
            }
            else{
                right++;
            }
        }
        return res;
        
    }
};